Userland/Services/SpiceAgent/Message.cpp

```diff
@@ -192,4 +192,62 @@ ErrorOr<String> ClipboardMessage::debug_description()
     return builder.to_string();
 }
 
+ErrorOr<FileTransferStartMessage> FileTransferStartMessage::read_from_stream(AK::Stream& stream)
+{
+    auto id = TRY(stream.read_value<u32>());
+
+    auto metadata_bytes = TRY(stream.read_until_eof());
+    auto metadata_content = TRY(String::from_utf8(metadata_bytes));
+
+    // TODO: We need some sort of INIParser, or we need to make Core::ConfigFile not depend on having an actual Core::File
+    // The first line in the file should always be `[vdagent-file-xfer]`
+    auto lines = TRY(metadata_content.split('\n'));
+    if (lines.is_empty() || lines.at(0) != "[vdagent-file-xfer]") {
+        return Error::from_string_literal("Failed to parse file transfer metadata");
+    }
+
+    String name {};
+    u32 size = -1;
```

Comment:
:/ `Optional<u32> size`

Reply:
Looks like it got dropped when I was messing around with my rebase. Will re-add it soon!